Complete Tracks by Length (nee. track accuracy over time) (#331)
Janelia-Trackathon-2023/traccuracy: docs/source/metrics/complete-tracks-by-length.md

(complete-tracks-by-length-metric)=
# Complete Tracks by Length

Complete Tracks by Length generalizes the CTC-BIO [Complete Tracks](complete-tracks) metric to every track length. For each length N (in frames), it measures the accuracy — the fraction of fully correct segments — of the ground truth tracklet or lineage segments that span N frames.

This is computed by sliding a window of N frames over the full extent of the dataset.
For each window, we count the correct and total ground truth tracklet/lineage segments within that window.
A tracklet/lineage segment is counted as correct if there are no errors anywhere in it. This includes checking:
- All edges are true positives (edge TP implies endpoint nodes are TP)
- There are no false positive divisions matched to this ground truth segment within the window

> Then, we sum the correct and total segments over all sliding windows, and divide the totals to get the overall accuracy for that length.

Reviewer: ```suggestion
Then, we sum the correct and total segments over all sliding windows of the given length, and divide the totals to get the overall accuracy for each window.
```

> For length 1, this is approximately the same as the fraction of correct edges (minus segments counted as incorrect due to false positive divisions).

Reviewer: ```suggestion
For a given length of 1, this is approximately the same as the fraction of correct edges (minus segments counted as incorrect due to false positive divisions).
```

> When the length equals the total extent of the data, this is the same as the CTC-BIO [Complete Tracks](complete-tracks).

Reviewer: ```suggestion
When the given length equals the number of frames in the data, this is the same as the CTC-BIO [Complete Tracks](complete-tracks) (as there is only a single window, and it spans all frames). 
```

> This metric returns the tracklet or lineage accuracy for all lengths from 1 to the maximum length, allowing you to see how the percentage of fully correct tracks degrades as the length you consider increases.

Reviewer: ```suggestion
This metric returns the tracklet or lineage accuracy for all lengths from 1 to the given length, allowing you to see how the percentage of fully correct tracks degrades as the length you consider increases.
```

> In lineage mode, if a division falls within the window, all branches are included in the same segment.

Reviewer: ```suggestion
In lineage mode, if a division falls within the currently evaluated window, all branches are included in the same segment.
```
